osidb/serializer.py

```diff
@@ -536,8 +537,18 @@ def get_meta(self, obj):
         serializer = MetaSerializer(instance=meta, many=True)
         return serializer.data
 
+    from django.db import transaction
+
+    @transaction.atomic
     def create(self, validated_data):
-        return Flaw.objects.create(**validated_data)
+        affects = validated_data.pop("affects", [])
+        flaw = Flaw(**validated_data)
+        flaw.save(raise_validation_error=False)
```

Comment:
I am not very familiar with the serializers so maybe this question is silly :smile: but does this allow the user to create a flaw without it being validated? That would not be correct. Or if it is something different, could you please explain to me what is this suppressed error raising necessary for?

Reply:
I'm not even sure that this works because nested writes (i.e. writing affects via the /flaws endpoint) is [not supported by default in DRF](https://www.django-rest-framework.org/api-guide/relations/#writable-nested-serializers) and here `affects` is not a list of `Affect` objects, in fact it isn't anything because `affects` is a read-only field meaning that it will never be found in `validated_data`

This needs some more thought, the solution isn't trivial so you may skip the endpoint tests that fail for now with the `pytest.mark.skip` decorator

Reply:
I agree, we can have a separated task to create the nested writes, I will restore the files related to serialization and submit it again with the 2 creation tests skiped.